yt/frontends/flash/io.py

```diff
@@ -192,30 +192,35 @@ def _yield_coordinates(self, data_file, needed_ptype=None):
         )
         yield ("io", pxyz)
 
-    def _read_particle_fields(self, chunks, ptf, selector):
-        chunks = list(chunks)
-        data_files = set()
-        assert len(ptf) == 1
-        for chunk in chunks:
-            for obj in chunk.objs:
-                data_files.update(obj.data_files)
+    def _read_data_file(self, data_file, ptf, selector=None):
```

Comment:
I immediately had anxiety about the routine name. What do you think about, for now at least, changing this to more explicitly be about particle data files? If it's in the particle io handler and not the base it might be okay not to rename.

Reply:
yes, totally up for a name change. especially since it gets called from `_read_particle_fields`. would be much clearer I think. `_read_particle_data_file`? Or maybe `_read_particle_range` to emphasize that we're selecting index ranges from a file (one of the frontends used `sub_files` instead of `data_files` to specify that)?

Reply:
`_read_from_particle_data_file`?

Reply:
went with `_read_particle_data_file`, but happy to change it!

Reply:
I like this solution